src/data/blob.rs

```diff
@@ -0,0 +1,420 @@
+
+        Self {
+            address,
+            value,
+            owner,
+        }
+    }
+
+    /// Returns the value.
+    pub fn value(&self) -> &Vec<u8> {
+        &self.value
+    }
+
+    /// Returns the set of owners.
+    pub fn is_owner(&self, user: PublicKey) -> bool {
+        self.owner == user
+    }
+
+    /// Returns the set of owners.
+    pub fn owner(&self) -> &PublicKey {
+        &self.owner
+    }
+
+    /// Returns the address.
+    pub fn address(&self) -> &Address {
+        &self.address
+    }
+
+    /// Returns the name.
+    pub fn name(&self) -> &XorName {
+        self.address.name()
+    }
+
+    /// Returns size of contained value.
+    pub fn payload_size(&self) -> usize {
+        self.value.len()
+    }
+
+    /// Returns size of this BlobData after serialisation.
+    pub fn serialised_size(&self) -> u64 {
+        serialized_size(self).unwrap_or(u64::MAX)
+    }
+
+    /// Return true if the size is valid
+    pub fn valid_size(&self) -> bool {
+        self.serialised_size() <= MAX_BLOB_SIZE_IN_BYTES
+    }
+}
+
+impl Serialize for PrivateBlob {
+    fn serialize<S: Serializer>(&self, serialiser: S) -> Result<S::Ok, S::Error> {
+        (&self.value, &self.owner).serialize(serialiser)
+    }
+}
+
+impl<'de> Deserialize<'de> for PrivateBlob {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        let (value, owner): (Vec<u8>, PublicKey) = Deserialize::deserialize(deserializer)?;
+        Ok(PrivateBlob::new(value, owner))
+    }
+}
+
+impl Debug for PrivateBlob {
+    fn fmt(&self, formatter: &mut Formatter) -> fmt::Result {
+        // TODO: Output owners?
+        write!(formatter, "PrivateBlob {:?}", self.name())
+    }
+}
+
+/// An immutable chunk of data which cannot be deleted.
+#[derive(Hash, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub struct PublicBlob {
+    /// Network address. Omitted when serialising and calculated from the `value` when
+    /// deserialising.
+    address: Address,
+    /// Contained data.
+    value: Vec<u8>,
+}
+
+impl PublicBlob {
+    /// Creates a new instance of `PublicBlob`
+    pub fn new(value: Vec<u8>) -> Self {
+        Self {
+            address: Address::Public(XorName(tiny_keccak::sha3_256(&value))),
+            value,
+        }
+    }
+
+    /// Returns the value.
+    pub fn value(&self) -> &Vec<u8> {
+        &self.value
+    }
+
+    /// Returns the address.
+    pub fn address(&self) -> &Address {
+        &self.address
+    }
+
+    /// Returns the name.
+    pub fn name(&self) -> &XorName {
+        self.address.name()
+    }
+
+    /// Returns size of contained value.
+    pub fn payload_size(&self) -> usize {
+        self.value.len()
+    }
+
+    /// Returns size of this BlobData after serialisation.
+    pub fn serialised_size(&self) -> u64 {
+        serialized_size(self).unwrap_or(u64::MAX)
+    }
+
+    /// Return true if the size is valid
+    pub fn valid_size(&self) -> bool {
```

Comment:
It's standard practice for boolean functions to start with a question word and it is a bit more clear when using such functions. So here I would name the function something like `has_valid_size`.

Reply:
I am aware of that. I have never used any other style (the rest of the data types consistently use **is_**, **has_** etc.) so was trying to remember why this got it. The only thing I can come think of is that this is the very first part I started (Blob), and I saw short forms of things I normally see prefixed with intention - like get; `value()` instead of `get_value()`, or` permissions(`) instead of `get_permissions()`. So I think that came as I perceived a code style there that I just didn't want to change too much.

As I got into the other types, there was already so much fixing that such a change seemed small.